code/ui/blocks/src/blocks/Canvas.tsx

```diff
@@ -99,10 +99,26 @@ const useDeprecatedPreviewProps = (
     children,
     layout: layoutProp,
     ...props
-  }: DeprecatedCanvasProps & { of?: ModuleExport; layout?: Layout },
+  }: DeprecatedCanvasProps & CanvasProps,
   docsContext: DocsContextProps<Renderer>,
   sourceContext: SourceContextProps
 ) => {
+  /* eslint-disable no-param-reassign */
+  mdxSource = mdxSource || props?.source?.code;
+  withSource =
+    withSource ||
+    (() => {
+      switch (props.sourceState) {
+        case 'none':
+          return SourceState.NONE;
+        case 'shown':
+          return SourceState.OPEN;
+        default:
+          return SourceState.CLOSED;
+      }
+    })();
+  /* eslint-enable no-param-reassign */
```

Comment:
I don't think it's safe to mutate the props here, you should save it as a new variable instead.